eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php

```diff
@@ -310,20 +310,55 @@ public function copySubtree($sourceId, $destinationParentId)
             );
         }
 
-        // If subtree root is main location for its content, update subtree section to the one of the
-        // parent location content
-        $subtreeRootContentInfo = $this->contentHandler->loadContentInfo($copiedSubtreeRootLocation->contentId);
-        if ($subtreeRootContentInfo->mainLocationId === $copiedSubtreeRootLocation->id) {
-            $this->setSectionForSubtree(
-                $copiedSubtreeRootLocation->id,
-                $this->contentHandler->loadContentInfo($this->load($destinationParentId)->contentId)->sectionId
-            );
-        }
+        $destinationParentSectionId = $this->getSectionId($destinationParentId);
+        $this->updateSubtreeSectionIfNecessary($copiedSubtreeRootLocation, $destinationParentSectionId);
 
         return $copiedSubtreeRootLocation;
     }
 
     /**
+     * Retrieves section ID of the location's content.
+     *
+     * @param int $locationId
+     *
+     * @return int
+     */
+    private function getSectionId($locationId)
+    {
+        $location = $this->load($locationId);
+        $locationContentInfo = $this->contentHandler->loadContentInfo($location->contentId);
+
+        return $locationContentInfo->sectionId;
+    }
+
+    /**
+     * If the location is the main location for its content, updates subtree section.
+     *
+     * @param Location $location
+     * @param int $sectionId
+     */
+    private function updateSubtreeSectionIfNecessary(Location $location, $sectionId)
+    {
+        if ($this->isMainLocation($location)) {
+            $this->setSectionForSubtree($location->id, $sectionId);
```

Comment:
Ok for the PR. For the legacy doc, it might mention what @natanael89 found: "If the sections of the source node and the destination node do match, then the subtree would be untouched, meaning it could be anything."